packages/devtools/lib/src/server_api_client.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:html';
+
+import 'package:sse/client/sse_client.dart';
+
+class ServerApiClient {
+  ServerApiClient() : _channel = SseClient('/api/sse') {
+    _channel.onOpen.first.then((_) {
+      // We'll want to send notifications when a user tries to reuse an existing
+      // DevTools window, so pop up the notification dialog early while they're
+      // looking at the page (this won't pop up if they've already answered).
+      Notification.requestPermission();
+    });
+
+    _channel.stream.listen((msg) {
+      try {
+        final request = jsonDecode(msg);
+        switch (request['method']) {
+          case 'connectToVm':
+            connectToVm(request['params']);
+            return;
+          case 'notify':
+            notify();
+            return;
+          default:
+            print('Unknown request ${request.method} from server');
+        }
+      } catch (e) {
+        print('Failed to handle API message from server:\n\n$msg\n\n$e');
+      }
+    });
+  }
+
+  final SseClient _channel;
+
+  int _nextRequestId = 0;
+  void _send(String method, [dynamic params]) {
```

Comment:
can you make params a `Map<String, dynamic>`?

Reply:
Yep!